### PR TITLE
verify a PowerShell terminal

### PR DESCRIPTION
**Pull request by elahehrashedi**
bugfix: 
https://github.com/microsoft/vscode-cmake-tools/issues/2650
https://github.com/microsoft/vscode-cmake-tools/issues/2621
https://github.com/microsoft/vscode-cmake-tools/issues/535

### CODE REVIEW

File: src/cmakeTools.ts [hunk 1]
```diff
@@ -2220,14 +2220,17 @@ export class CMakeTools implements api.CMakeToolsAPI {
...
 
         let executablePath = shlex.quote(executable.path);
 
         if (process.platform === 'win32') {
             executablePath = executablePath.replace(/\\/g, "/");
-
-            if (vscode.env.shell.indexOf("PowerShell") > 0) {
+            const launchBehavior = this.workspaceContext.config.launchBehavior.toLowerCase();
+            const launchTerminalPath = (terminal.creationOptions as vscode.TerminalOptions).env![this.launchTerminalPath];
+            if (vscode.env.shell.indexOf("PowerShell") > 0
```
Comment (bobbrow) [inline src/cmakeTools.ts:2231]: The indexof checks should always be against `launchTerminalPath`, not `vscode.env.shell`.
We should do case-insensitive checks, so please lowercase the path first.
We shouldn't need to do anything different based on the `launchBehavior`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ Bug Fixes:
 - Clear output channel after auto-reconfigure. [#2628](https://github.com/microsoft/vscode-cmake-tools/issues/2628)
 - Don't delete CMakeCache.txt when switching kits if the buildDirectory also changes. [#2546](https://github.com/microsoft/vscode-cmake-tools/issues/2546) [@david-fong](https://github.com/david-fong)
 - Add "description" properties to the cmake.revealLog setting. [#2578](https://github.com/microsoft/vscode-cmake-tools/issues/2578)
+- Support launching the target in new powershell terminal. [#2650](https://github.com/microsoft/vscode-cmake-tools/issues/2650)
```
Comment (bobbrow) [inline CHANGELOG.md:15]: "Fix issues with launching targets in PowerShell"

Comment (bobbrow) [reply CHANGELOG.md:15]: And add the other issues here too.  2621, 535

File: src/cmakeTools.ts [hunk 3]
```diff
@@ -2220,14 +2220,14 @@ export class CMakeTools implements api.CMakeToolsAPI {
...
+        const terminal: vscode.Terminal = await this.createTerminal(executable);
 
         let executablePath = shlex.quote(executable.path);
 
         if (process.platform === 'win32') {
             executablePath = executablePath.replace(/\\/g, "/");
-
-            if (vscode.env.shell.indexOf("PowerShell") > 0) {
+            const launchTerminalPath = (terminal.creationOptions as vscode.TerminalOptions).env![this.launchTerminalPath];
+            if (launchTerminalPath?.toLocaleLowerCase().includes("pwsh.exe")) {
```
Comment (bobbrow) [inline src/cmakeTools.ts:2230]: We also need to check for "powershell"

**Review by bobbrow — APPROVED** [review]

File: src/cmakeTools.ts [hunk 4]
```diff
@@ -2220,14 +2220,14 @@ export class CMakeTools implements api.CMakeToolsAPI {
...
+        const terminal: vscode.Terminal = await this.createTerminal(executable);
 
         let executablePath = shlex.quote(executable.path);
 
         if (process.platform === 'win32') {
             executablePath = executablePath.replace(/\\/g, "/");
-
-            if (vscode.env.shell.indexOf("PowerShell") > 0) {
+            const launchTerminalPath = (terminal.creationOptions as vscode.TerminalOptions).env![this.launchTerminalPath]?.toLocaleLowerCase();
+            if (launchTerminalPath?.includes("pwsh.exe") || launchTerminalPath?.includes("powershell")) {
```
Comment (bobbrow) [inline src/cmakeTools.ts:2230]: We need to check for `pwsh` on linux/mac

**Comment by elahehrashedi** [comment]
@bobbrow made a small change. This is tested on WSL. Do you want to take a look before merging?

**Review by bobbrow — APPROVED** [review]

**Pull request merged**